### PR TITLE
*: fix the Host column of 'show processlist'

### PR DESCRIPTION
**Pull request by wshwsh12**
### What problem does this PR solve? 
#10903 
The `Host` column should be `host_name:client_port` format.

### What is changed and how it works?
open 2 mysql client and run `show processlist;` both.
Before this pr:
```
mysql> show processlist;
+------+------+-----------+------+---------+------+-------+------------------+
| Id   | User | Host      | db   | Command | Time | State | Info             |
+------+------+-----------+------+---------+------+-------+------------------+
|    1 | root | 127.0.0.1 | NULL | Sleep   |    6 | 2     | NULL             |
|    2 | root | 127.0.0.1 | NULL | Query   |    0 | 2     | show processlist |
+------+------+-----------+------+---------+------+-------+------------------+
2 rows in set (0.00 sec)

```

This pr:
```
mysql> show processlist;
+------+------+-----------------+------+---------+------+-------+------------------+
| Id   | User | Host            | db   | Command | Time | State | Info             |
+------+------+-----------------+------+---------+------+-------+------------------+
|    1 | root | 127.0.0.1:49032 | NULL | Sleep   |   10 | 2     | NULL             |
|    2 | root | 127.0.0.1:49034 | NULL | Query   |    0 | 2     | show processlist |
+------+------+-----------------+------+---------+------+-------+------------------+
2 rows in set (0.00 sec)


```

### Check List 

Tests 

 - Unit test
 - Manual test (add detailed scripts or steps below)

Code changes

 - Has exported function/method change

Side effects

 - Increased code complexity

Related changes

### CODE REVIEW

File: session/session.go [hunk 1]
```diff
@@ -990,6 +990,9 @@ func (s *session) SetProcessInfo(sql string, t time.Time, command byte, maxExecu
 	if s.sessionVars.User != nil {
 		pi.User = s.sessionVars.User.Username
 		pi.Host = s.sessionVars.User.Hostname
+		if s.sessionVars.ConnectionInfo != nil {
```
Comment (XuHuaiyu) [inline session/session.go]: Can we put this check outside of `if s.sessionVars.User != nil {`?

**Comment by lysu** [comment]
hi @wshwsh12 , `ConnectionInfo` should better not use in Non-plugin function... that structure maybe contain more info fields and be more heavy in future.

how about add a new `ClientConn` interface and make `server.clientConn` impl that interface then save it in `SessionVars`? current question is we can not get clientConn from session?

**Pull request closed without merging**